resources/assets/lib/select-options.coffee

```diff
@@ -0,0 +1,119 @@
+###
+#    Copyright 2015-2018 ppy Pty. Ltd.
+#
+#    This file is part of osu!web. osu!web is distributed with the hope of
+#    attracting more community contributions to the core ecosystem of osu!.
+#
+#    osu!web is free software: you can redistribute it and/or modify
+#    it under the terms of the Affero GNU General Public License version 3
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+import { a, i, div } from 'react-dom-factories'
+import { PureComponent } from 'react'
+
+export class SelectOptions extends PureComponent
+  constructor: (props) ->
+    super props
+    @bn = @props.bn ? 'select-options'
+
+    @state =
+      showingSelector: false
+
+
+  componentDidMount: =>
+    $(document).on "click.#{@bn}", @hideSelector
+
+
+  componentDidUpdate: (_prevProps, prevState) =>
+    Blackout.toggle(@state.showingSelector, 0.5) unless prevState.showingSelector == @state.showingSelector
+
+
+  componentWillUnmount: ->
+    $(document).off ".#{@bn}"
+
+
+  render: =>
+    classNames = "#{@bn} js-#{@bn}"
```

Comment:
make a ref instead so can just refer that directly instead of using `js-#{@bn}` class name.